content/riak/kv/2.2.0/using/performance/open-files-limit.md

```diff
@@ -15,326 +15,278 @@ aliases:
-the file:
+You can enable PAM-based user limits so that non-root users, such as the `riak` user, may specify a higher value for maximum open files.
 
-    <div><pre><code>&ast;              soft     nofile          200000
-    &ast;              hard     nofile          200000
-    </code></pre></div>
+For example, follow these steps to enable PAM-based limits for all users to allow a maximum of 200000 open files.
 
-2\. Save and close the file.
+1\. Edit /etc/pam.d/login and add the following line:
 
-3\. Restart the machine so that the limits to take effect and verify that
+```/etc/pam.d/login
+session    required   pam_limits.so
+```
+
+2\. Save and close /etc/pam.d/login
+
+3\. Edit /etc/security/limits.conf and append the following lines to the file:
+
+```/etc/security/limits.conf
+* soft nofile 65536
+* hard nofile 200000
+```
+
+4\. Save and close the /etc/security/limits.conf file.
+
+5\. Restart the machine so that the limits to take effect and verify that
 the new limits are set with the following command:
 
 ```bash
 ulimit -a
 ```
 
 {{% note %}}
-In the above examples, the
-open files limit is raised for all users of the system. If you prefer,
-the limit can be specified for the riak user only by substituting the
-two asterisks (`*`) in the examples with `riak<`.
+In the above examples, the open files limit is raised for all users of the system. The limit can be specified for the riak user only by substituting the
+two asterisks (`*`) in the examples with `riak`.
 {{% /note %}}
 
 
 ## Solaris
 
-In Solaris 8, there is a default limit of 1024 file descriptors per
-process. In Solaris 9, the default limit was raised to 65536. To
-increase the per-process limit on Solaris, add the following line to
-`/etc/system`:
+To increase the open file limit on Solaris, add the following line to the /etc/system file:
 
-```config
+```/etc/system
 set rlim_fd_max=200000
 ```
 
-[Reference](http://blogs.oracle.com/elving/entry/too_many_open_files)
+[Reference][blog oracle]
+
+## Mac OSX El Capitan
+
+Start by checking the current open file limit values with:
+
+```bash
+launchctl limit maxfiles
+```
+
+The response should looking something like this:
+
+```bash
+maxfiles    65536          65536
+```
+
+The first column is the soft limit and the last column is the hard limit.
 
-## Mac OS X
+To change the open files limits on Mac OS X Yosemite, perform the following steps:
 
-To check the current limits on your Mac OS X system, run:
+1\. Add the following line to your .bash_profile or analogous file:
+
+```bash
+ulimit -n 65536 200000
+```
+
+2\. Save and close the file. Next open /etc/sysctl.conf (or create it if it doesn't already exist) and add the following settings:
+
+```/etc/sysctl.conf
+kern.maxfiles=200000
+kern.maxfilesperproc=200000
+```
+
+4\. Restart your computer and enter `ulimit -n` into your terminal. If your system is configured correctly, you should see that `maxfiles` has been set to 200000.
+
+
+## Mac OSX Yosemite
```

Comment:
Maybe this just looks weird in the diff, but why is this heading down here when the instructions are  up there ^?